stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -31,6 +31,16 @@
  * not need to be synchronized together; incoming and outgoing directions are independent.
  * Since individual {@code StreamObserver}s are not thread-safe, if multiple threads will be
  * writing to a {@code StreamObserver} concurrently, the application must synchronize calls.
+ *
+ * <p>This API is asynchronous, so methods may return before the operation completes. The API
+ * provides no guarantees for how quickly an operation will complete, so utilizing flow control via
+ * {@link ClientCallStreamObserver} and {@link ServerCallStreamObserver} to avoid excessive
+ * buffering is recommended for streaming RPCs. gRPC's implementation of {@code onError()} on
+ * client-side causes the RPC to be cancelled and discards all messages, so completes quickly.
```

Comment:
"so completes quickly" - How quickly is quick? Does this behavior guarantee anything users can rely on?

Reply:
No, we aren't going to define how quickly. I think it is helpful to understand the system and what things do. And it is also useful to state this behavior applies only to client-side. If a user saw client-side cleaned up quickly, this documentation would define that behavior and tell them to not assume server-side is the same.